### PR TITLE
Fix personal vocab edit order, learner stats and add esc functionality

### PR DESCRIPTION
**Pull request by Tanvez**
This PR fixes the order of personal vocab when the user edits the headword, the learner familiarity stats for a given lemmatized text, add esc functionality.
- on delete of a personal vocab entry will trigger a stat update for all lemmatized text that has stats with a corresponding vocab list
- order of the edited personal vocab entry will be preserved upon save via `actions.js`
- when editing a row, when esc is pressed it will exit out of the edit feature

Note:
- there are some styling changes in the vue component
- order of the personal vocab entry is preserved by mutating only the entry in the local personalVocabList state using the data returned from the backend. This is assuming that there are no new personal vocab entries being added simultaneously otherwise the data would be stale.

### BUGS/TODO
- overflow-x causes the table rows to overflow in mobile view but keeps tooltip overlay in desktop view

### CODE REVIEW

File: static/src/js/app/vuex/actions.js [hunk 1]
```diff
@@ -59,7 +59,14 @@ export default {
     if (response && response.status >= 400) {
       return response;
     }
-    commit(FETCH_PERSONAL_VOCAB_LIST, data);
+    const { entries } = data;
```
Comment (Tanvez) [inline static/src/js/app/vuex/actions.js:62]: TODO: Add comment about why- preserving order

**Review by jaguillette — APPROVED** [review]
Looked at the changes since our review yesterday, and it looks great!

File: static/src/js/app/vuex/actions.js [hunk 2]
```diff
@@ -59,7 +59,14 @@ export default {
...
+    const { entries: localEntries } = state.personalVocabList;
+    const foundObj = entries.find((el) => el.id === entryId);
+    const localEntryIndex = localEntries.findIndex((el) => el.id === entryId);
+    const updatedEntries = { ...state.personalVocabList };
+    updatedEntries.entries[localEntryIndex] = foundObj;
+
+    commit(FETCH_PERSONAL_VOCAB_LIST, updatedEntries);
```
[the remark this answers is not in the document]
Comment (Tanvez) [reply static/src/js/app/vuex/actions.js:75]: Yes, that is correct! Essentially the code is making a copy of the existing `personal vocab list` vuex state => assigning the newly updated entry to the specific entry in the array/list  => committing the new list to vuex state

**Review by arthurian — APPROVED** [review]
The changes look great! Nice work 👍

**Pull request merged**